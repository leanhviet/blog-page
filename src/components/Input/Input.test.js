// Libs
import React from 'react'
import { shallow } from 'enzyme'

// Components
import { InputBase } from './index'

const props = {
  type: 'text',
  defaultValue: 'Search',
  placeholder: 'Search...',
  className: 'input',
}

describe('InputBase component', () => {
  const wrapper = shallow(<InputBase />)
  const wrapperWithProps = shallow(<InputBase {...props} />)

  it('should be defined InputBase component', () => {
    expect(wrapper).toBeDefined()
  })

  // Testing Snapshot
  it('should be render snapshot InputBase component', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapperWithProps).toMatchSnapshot()
  })

  // Testing default props
  it('should be render defaultProps InputBase component', () => {
    expect(InputBase.defaultProps.className).toEqual('')
    expect(InputBase.defaultProps.type).toEqual('text')
    expect(InputBase.defaultProps.defaultValue).toEqual('')
    expect(InputBase.defaultProps.placeholder).toEqual('')
  })

  // Testing mock props
  it('should be render mockProps InputBase component', () => {
    expect(wrapperWithProps.find('input').props().className).toContain(
      props.className,
    )
    expect(wrapperWithProps.find('input').props().placeholder).toEqual(
      props.placeholder,
    )
    expect(wrapperWithProps.find('input').props().type).toEqual(props.type)
    expect(wrapperWithProps.find('input').props().value).toEqual(
      props.defaultValue,
    )
  })
})
