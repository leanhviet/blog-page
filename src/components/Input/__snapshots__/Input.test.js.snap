// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBase component should be render snapshot InputBase component 1`] = `
<div
  className="input-group"
>
  <div
    className="input-group__field"
  >
    <input
      className="input-group__field__text  "
      onChange={[Function]}
      placeholder=""
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`InputBase component should be render snapshot InputBase component 2`] = `
<div
  className="input-group"
>
  <div
    className="input-group__field"
  >
    <input
      className="input-group__field__text  input"
      onChange={[Function]}
      placeholder="Search..."
      type="text"
      value="Search"
    />
  </div>
</div>
`;
